### ISSUE TITLE
XUnit test works well but Program.cs run yields terrible results

### ISSUE THREAD

**Issue by tienle488** [PragmaticFlow/NBomber#333]
Hi.

This is the 2nd project that I've worked on using NBomber load testing but the model that is working and I'm following should work for this 2nd project; however, results are bad and not as expected.

Tests that produce acceptable results will have to come from running my XUnit test instead.  Reference attachment(s).

Any insight is appreciated.

Thanks,
Tien
![csep1593 1200secsProgram csRunWithXUnitTestOf NETCoreCryptoClient infrastructure evenWorseResults](https://user-images.githubusercontent.com/23621965/111000847-a505e900-8337-11eb-96b2-d9810841e3c1.PNG)

**Comment by AntyaDev** [comment]
Hi @tienle488 
Thanks for posting this issue.
How I can simulate this test?

**Comment by tienle488** [comment]
Hello @AntyaDev 

I will put something together so that you can hopefully reproduce the problem from your end.  Thanks!
Tien

**Comment by AntyaDev** [comment]
Hi @tienle488 
Ok, no problem.

Is this problem is relative to this one?
https://github.com/PragmaticFlow/NBomber/issues/316#issuecomment-796528053

**Comment by tienle488** [comment]
No, @Antya ***@***.***>.  This is a different problem.
________________________________

**Comment by tienle488** [comment]
Hello @AntyaDev 
![Program csRunVsXUnitTestRun_badVsGoodRespectively_SimpleHttpTest NBomber 2ndTest](https://user-images.githubusercontent.com/23621965/111546844-85a2fd80-8735-11eb-913f-7ca191e78fdc.PNG)


I ran the SimpleHttpTest from the XUnitTest of NBomber solution I cloned from GitHub and the results look correct (100 requests per second for 30 secs returned 3000 requests).  I also ran the load test with SimpleHttpTest from Program.cs and the results were bad like I reported with my test scenario.  I did this twice so that the results are confirmed.  Reference attachment.  If I can be of further assistance, please let me know.  Thanks, Antya.

Tien

**Comment by AntyaDev** [comment]
Hi @tienle488 
I see that you run your test under the old version of NBomber.
Can you please re-run your test using new version

**Comment by tienle488** [comment]
Hello @AntyaDev 
Looks like with latest NBomber solution, the problem is not seen (
![NBomberV2 0 0-Beta6LooksGoodForIssue333](https://user-images.githubusercontent.com/23621965/112076636-039d4500-8b38-11eb-8580-1fe8855cad73.PNG)
**status_stats branch**
); however, when I tried running the scenario with my load test solution after updating the NBomber NuGet to v2.0.0-beta6, I'm encountering what I reported earlier with issue #316 when I updated NBomber to beta version with my Telemetry Client test.  With my Crypto client tests:   
My XUnit test is taking 9mins to complete instead of the 10secs test.  The results are bad as I reported previously:
![NBomberV2 0 0-beta6InB2bPlatformLoadTests_XUnitTestRunsTooLongAndBadResults](https://user-images.githubusercontent.com/23621965/112076881-81615080-8b38-11eb-8fa9-cb356a0a19bc.PNG)

When I try running with Program.cs, I'm getting the similar results:  12mins to produce results after the 10secs test
![NBomberV2 0 0-beta6InB2bPlatformLoadTests_Program csTestRunsTooLongAndBadResultsAsWell](https://user-images.githubusercontent.com/23621965/112076966-ac4ba480-8b38-11eb-83d3-3461690f482b.PNG)

Is there a NuGet package I need to add?
![NBomberV2 0 0-beta6InB2bPlatformLoadTests_testsFail_missingPackagePossibly](https://user-images.githubusercontent.com/23621965/112076992-bc638400-8b38-11eb-95af-4f72ed011aa7.PNG)

Thanks in advance for your help,
Tien
